includes/polling/ipmi.inc.php

```diff
@@ -2,56 +2,65 @@
 
 $ipmi_rows = dbFetchRows("SELECT * FROM sensors WHERE device_id = ? AND poller_type='ipmi'", array($device['device_id']));
 
-d_echo($ipmi_rows);
+if (is_array($ipmi_rows)) {
+    d_echo($ipmi_rows);
 
-if ($ipmi['host'] = get_dev_attrib($device, 'ipmi_hostname')) {
-    $ipmi['user']     = get_dev_attrib($device, 'ipmi_username');
-    $ipmi['password'] = get_dev_attrib($device, 'ipmi_password');
-    $ipmi['type']     = get_dev_attrib($device, 'ipmi_type');
-    echo 'Fetching IPMI sensor data...';
+    if ($ipmi['host'] = $attribs['ipmi_hostname']) {
```

Comment:
I'll leave that refactor to another time but yes it's a good idea as at present we just get them all at the start of the poll.